Ports Byond 510 preperations from Bay

Yoshax opened the pull request:
Ports Baystation12/Baystation12#12243

Along with preparing for 510, allows for easier porting of newer bay PRs and suchlike.

Datraen commented:
Doesn't contain all of the files changed by the Bay preparation, and has compiling errors.

Yoshax commented:
Added 1 missing file, the other is unecessary as our codebases differ there. It's: code/modules/examine/descriptions/armor.dm

They previously returned, return list2text(.) and changed it to, return jointext(., null)
However, we return a var.

Additonally, compiles fine for me on 510 and 509, see:
510: https://i.gyazo.com/93b0b60fde1826212c87d75f6168d571.png
509: https://i.gyazo.com/d4a5f24b4a64136180256cec36257da0.png

Datraen commented:
Not sure why I put the Needs Testing tag on.
I'm just holding off on merging anything more today, since I'm a little sleep deprived.
